### DISCUSSION TITLE
Problem with args.Player.Teleport

### DISCUSSION THREAD

**Discussion by LuanIllogical** [Pryaxis/TShock#2102] [Q&A] [answered]
No matter what numbers I put, even if I put Main.rand.Next, it will always teleport me to this specific spot, what am I doing wrong?

        void TestTeleport(CommandArgs args)
        {
            args.Player.Teleport(472, 111, 1);
        }

![image](https://user-images.githubusercontent.com/60335462/88605028-2d58a780-d04f-11ea-8a35-8c356c3915b9.png)

**Comment by Quinci135** [comment, marked as answer]
Tile coordinates are 1/16 of world coordinates (each tile is 16x16 pixels)
So you would be teleporting 472 pixels right 111 pixels down from the top left corner of the world (not very far)
https://github.com/tModLoader/tModLoader/wiki/Basic-Tile#coordinates

Multiply both coordinates by 16 to get world coordinates (by whole tile instead of pixels) which is used for npcs, players, and projectiles which you use for `args.Player.Teleport()`
eg `args.Player.Teleport(X * 16, Y * 16, 1);`

Had to do that to teleport a player to wherever they pinged on the map
https://github.com/Quinci135/MapTeleport/blob/master/MapTeleport/MapTeleport.cs#L55

**Reply by LuanIllogical** [reply]
I'm gonna try that, still doesn't explain why with random it still always went to that place but it might have been just bad luck, thanks!

**Reply by LuanIllogical** [reply]
It worked! thank you so much
